### PR TITLE
docs(api): add `addRouteMiddleware` util

### PR DESCRIPTION
**Pull request by Krutie**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Added API documentation for `addRouteMiddleware` utility function

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/3.api/3.utils/add-route-middleware.md [hunk 1]
```diff
@@ -1,7 +1,73 @@
 # `addRouteMiddleware`
 
-::ReadMore{link="/guide/features/routing"}
-::
+`addRouteMiddleware()` is a helper function to dynamically add route middleware in your Nuxt application.
+
+Route middleware are navigation guards and they are generally stored in the `/middelware` directory of your Nuxt application. But using `addRouteMiddleware()` utility function, we can add route middleware directly where they are needed, such as Nuxt plugins.
```
Comment (stenet) [inline docs/content/3.api/3.utils/add-route-middleware.md:5]: Should be "By using ..." instead of "But using ..."

Comment (Atinux) [inline docs/content/3.api/3.utils/add-route-middleware.md:5]:
```suggestion
::alert{type=info}
Route middleware are navigation guards and are generally stored in the [`middleware/`](/guide/directory-structure/middleware) directory of your Nuxt application.
::
```

File: docs/content/3.api/3.utils/add-route-middleware.md [hunk 2]
```diff
@@ -1,7 +1,75 @@
...
+
+You can also set  an optional, third argument `{ global: true }` to indicate whether the route middleware being defined is `global`.
+
+```js [plugins/my-plugin.ts]
+export default defineNuxtPlugin(() => {
+  addRouteMiddleware('global-middleware', (to, from) => {
+      console.log('global middleware that runs on every route change')
+    }, 
+ { global: true })
+})
```
Comment (DamianGlowala) [inline docs/content/3.api/3.utils/add-route-middleware.md:71]:
```suggestion
### Global Route Middleware

You can set an optional, third argument `{ global: true }` to indicate whether the route middleware is global:

```ts [plugins/my-plugin.ts]
export default defineNuxtPlugin(() => {
  addRouteMiddleware('global-middleware', (to, from) => {
      console.log('global middleware that runs on every route change')
    },
    { global: true }
  )
})
```

File: docs/content/3.api/3.utils/add-route-middleware.md [hunk 3]
```diff
@@ -1,7 +1,75 @@
...
+```
+
+### Named route middleware
+
+Named route middleware always takes the first argument as a `string` and the second argument as a `function`.
+
+Named route middleware defined through a plugin like this overrides any existing middleware of the same name located in the `/middleware` directory.
+
+```js [plugins/my-plugin.ts]
```
Comment (DamianGlowala) [inline docs/content/3.api/3.utils/add-route-middleware.md:53]:
```suggestion
### Named Route Middleware

Named route middleware takes `string` as the first argument and `function` as the second.

When defined in a plugin, it overrides any existing middleware of the same name located in the `/middleware` directory:

```ts [plugins/my-plugin.ts]
```

File: docs/content/3.api/3.utils/add-route-middleware.md [hunk 4]
```diff
@@ -1,7 +1,75 @@
...
+
+`options` lets you set the value of `global` to `true` to indicate whether the router middleware is global or not. By default the `global` is set to `false`.
+
+## Examples
+
+### Inline route middleware
+
+Inline route middleware is anonymous because it does not have a name. `function` is passed as the first argument and therefore the second argument of `middleware` becomes redundant.
+
+```js [plugins/my-plugin.ts]
```
Comment (DamianGlowala) [inline docs/content/3.api/3.utils/add-route-middleware.md:37]:
```suggestion
### Anonymous Route Middleware

Anonymous route middleware does not have a name. It takes `function` as the first argument, making the second `middleware` argument redundant:

```ts [plugins/my-plugin.ts]
```

File: docs/content/3.api/3.utils/add-route-middleware.md [hunk 5]
```diff
@@ -1,7 +1,75 @@
...
+addRouteMiddleware (name: string | RouteMiddleware, middleware?: RouteMiddleware, options: AddRouteMiddlewareOptions = {})
+```
+
+`addRouteMiddleware()` takes three arguments:
+
+- **name** `type: string | RouteMiddleware`
+
+`name` can either be a `string` or a function of type `RouteMiddleware`. This function provides the next route `to` as the first argument and the current route `from` as the second argument. Both `to` and `from` are Vue route objects.
+
+Learn more about available properties found on [route objects](/api/composables/use-route).
```
Comment (DamianGlowala) [inline docs/content/3.api/3.utils/add-route-middleware.md:21]:
```suggestion
`name` can be either a string or a function of type `RouteMiddleware`. Function takes the next route `to` as the first argument and the current route `from` as the second argument, both of which are Vue route objects.

Learn more about available properties of [route objects](/api/composables/use-route).
```

File: docs/content/3.api/3.utils/add-route-middleware.md [hunk 6]
```diff
@@ -1,7 +1,75 @@
...
+
+- **middleware**: `type: RouteMiddleware`
+
+The second argument is also a function of type `RouteMiddleware` but it is optional. Same as mentioned above, this function provides `to` and `from` route objects. This argument becomes optional if the first argument in `addRouteMiddleware()` is already passed as a function.
```
Comment (DamianGlowala) [inline docs/content/3.api/3.utils/add-route-middleware.md:25]:
```suggestion
The second argument is a function of type `RouteMiddleware`. Same as above, it provides `to` and `from` route objects. It becomes optional if the first argument in `addRouteMiddleware()` is already passed as a function.
```

File: docs/content/3.api/3.utils/add-route-middleware.md [hunk 7]
```diff
@@ -1,7 +1,75 @@
...
+
+- **options:** `type: AddRouteMiddlewareOptions`  
+
+`options` lets you set the value of `global` to `true` to indicate whether the router middleware is global or not. By default the `global` is set to `false`.
```
Comment (DamianGlowala) [inline docs/content/3.api/3.utils/add-route-middleware.md:29]:
```suggestion
An optional `options` argument lets you set the value of `global` to `true` to indicate whether the router middleware is global or not (set to `false` by default).
```

**Review by DamianGlowala — CHANGES_REQUESTED** [review]

File: docs/content/3.api/3.utils/add-route-middleware.md [hunk 8]
```diff
@@ -1,7 +1,76 @@
...
+    }
+  })
+})
+```
+
+### Named Route Middleware
+
+Named route middleware takes a string as the first argument and a function as the second.
+
+When defined in a plugin, it overrides any existing middleware of the same name located in the `/middleware` directory:
```
Comment (DamianGlowala) [inline docs/content/3.api/3.utils/add-route-middleware.md:51]:
```suggestion
When defined in a plugin, it overrides any existing middleware of the same name located in the `middleware/` directory:
```

**Review by DamianGlowala — APPROVED** [review]

**Review by pi0 — APPROVED** [review]
Thanks!

**Pull request merged**